nixos/modules/services/web-apps/ackee.nix

```diff
@@ -0,0 +1,131 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.ackee;
+  defaultSettings = {
+    ACKEE_MONGODB = "mongodb://localhost:27017/ackee";
+    ACKEE_USERNAME = "ackee";
+    ACKEE_PORT = "3000";
+    NODE_ENV = "production";
+    HOME = "%S/ackee";
+  };
+in
+{
+  options.services.ackee = {
+    enable = mkEnableOption "the Ackee analytics tool";
+
+    settings = mkOption {
+      type = with types; attrsOf (nullOr (oneOf [ str path package ]));
```

Comment:
This should probably be

```
type = with types; submodule {
  freeformType = attrsOf (nullOr (oneOf [ str path package ])));
};
```

instead. This _should_ get you the correct merging logic - then you shouldn't need to set `services.ackee.settings` below using the mapAttrs thing.

Reply:
I tried setting this `type`, but overwriting one of the settings still causes all of the defaults to disappear.